Test passwordless signup for new users

donlair opened the pull request:
#### Changes proposed in this Pull Request
This PR re-implements the passwordless signup, which will run as a test again the control experience. The test is configured to run via two Explat tests, one each for mobile and desktop viewports. Those test names are:

Desktop: `registration_email_only_desktop`
Mobile: `registration_email_only_mobile`

Context behind this test is at p58i-boI-p2#comment-52280.

Before:
<img width="894" alt="CleanShot 2021-11-10 at 16 27 04@2x" src="https://user-images.githubusercontent.com/35781181/141319021-5379680c-11db-4e3e-88b2-177140cef5c5.png">

After:
<img width="956" alt="CleanShot 2021-11-10 at 11 12 08@2x" src="https://user-images.githubusercontent.com/35781181/141319035-8bf718b8-ded3-4cbf-83ae-71ccc9048111.png">


#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Checkout this PR and start Calypso.
* Navigate to both tests in Explat.
* Add the `control` and `treatment` variation bookmarklets to your nav bar.
* Navigate to `/start/new` on your Calypso instance and assign yourself to the `treatment` variation.
* Repeat the tests for both desktop and mobile viewports.

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

donlair committed:
Passwordless signup initial commit

donlair committed:
Bug fix and style update

niranjan-uma-shankar commented on client/blocks/signup-form/index.jsx:
```diff
@@ -1013,6 +1016,59 @@ class SignupForm extends Component {
+						logInUrl={ logInUrl }
+						disabled={ this.props.disabled }
+						disableSubmitButton={ this.props.disableSubmitButton }
+						recaptchaClientId={ this.props.recaptchaClientId }
```
I think we can safely remove all of the recaptcha code in this file and in `client/blocks/signup-form/passwordless.jsx`.

niranjan-uma-shankar reviewed (commented):
1. The login link doesn't work if I use an existing email ID:
![eos](https://user-images.githubusercontent.com/1269602/141440744-231f28a8-ec43-4824-948d-a848affc85ff.gif)

2. When I click on the "Treatment" bookmarklet to assign myself to treatment, it takes about a minute or so to actually see the treatment UI. Until then, any number of page refreshes shows only the control version. Is this delay expected?

3. The login page shifts from white background to blue background when going back from the magic login page. Here're the steps to reproduce:
- In treatment, sign up for a new account, and log out.
- Go to /start/user
- Click on the "Log In" link in the sub-header
- Enter the email address with which you signed up in Step 1 to log in
- You'll be taken to the magic login page 
- Click the Back link, and you'll be taken to the blue login page.

![eosl](https://user-images.githubusercontent.com/1269602/141450589-ec918cfb-4490-4141-83aa-130568050e13.gif)

donlair committed:
Addressing review feedback.

donlair commented:
Good catches, thanks @niranjan-uma-shankar! This is ready for another look when you get a chance.

donlair committed:
Implementing separate tests for mobile andd desktop viewports

donlair commented:
> When I click on the "Treatment" bookmarklet to assign myself to treatment, it takes about a minute or so to actually see the treatment UI. Until then, any number of page refreshes shows only the control version. Is this delay expected?

The same thing has been happening to me w/ all Explat assignments made via the bookmarklet lately, so I don't think it's related to anything in this particular PR.

I split this this into two separate Explat tests per you comment at pbxNRc-144-p2#comment-2599. Thanks for the suggestion!

niranjan-uma-shankar commented on client/blocks/signup-form/index.jsx:
```diff
@@ -1013,10 +994,58 @@ class SignupForm extends Component {
+			We are testing whether a passwordless account creation and login improves signup rate in the `onboarding` flow
+		*/
+		if (
+			( this.props.flowName === 'onboarding' || this.props.flowName === 'test-fse' ) &&
```
Minor: We can remove the `test-fse` check since this flow is not used any longer.

niranjan-uma-shankar reviewed (commented):
1. On the mobile treatment experiment, I was redirected to the login page after the plans step. Could you please check if you're seeing this too?
2. I checked in non-EN and a couple of strings were appearing untranslated - the label for the email field and the error message when using an existing email address.
<img width="785" alt="Screenshot 2021-11-15 at 12 35 10 PM" src="https://user-images.githubusercontent.com/1269602/141749869-4f48d39c-972c-4527-8834-618ff2de8ece.png">
<img width="469" alt="Screenshot 2021-11-15 at 12 37 51 PM" src="https://user-images.githubusercontent.com/1269602/141749943-c84f57a7-6d2e-4978-b101-f368e02208aa.png">

donlair committed:
Copy tweak and removing reference to unused flow

donlair commented:
Rebased and addressed your feedback @niranjan-uma-shankar. Let me know if you have any thoughts on why that translated string isn't showing up as translated.

niranjan-uma-shankar commented:
> I saw the new welcome experience that asks if you want to start building or writing, which then took me to the Calypso dashboard. Are you being taken to the login page consistly?

I'm able to see this consistently but only for iOS devices, not Android. Can you please confirm if you tested with an iPhone device? The steps I followed:
* Using Chrome's "Toggle device" feature, I selected an iPhone X and went through the signup flow.
* I clicked on "View plans" in the domain step, and selected a Premium plan on the plans step.
* I got redirected to the login page at this point.

I think this might be related to https://github.com/Automattic/wp-calypso/issues/58061

> I checked translate.wordpress.com and "Enter your email address" is a translated string, so I'm surprised it's showing the untranslated text

The string is indeed translated, but all of its usage seems to be on the backend so looks those translations are not available on Calypso. I'd suggest checking in with Team Global in Slack #i18n channel about how to import those over here.

niranjan-uma-shankar commented:
> I think this might be related to #58061

If you are able to try the calypso.live link in a physical iPhone and you are able to proceed to the checkout page from the plans step, then let's move ahead. Probably what I saw was related to the iOS bug.

I see a brief flicker of the user step after clicking on "Create account" that says "Your account has already been created ..."

![eost1](https://user-images.githubusercontent.com/1269602/141990380-85b502fe-e9f5-4fa7-a27c-a54a87db970e.gif)



![eost2](https://user-images.githubusercontent.com/1269602/141990565-b7376422-3af4-41cb-a4e8-68fe92e913ce.gif)

It may probably not affect the experiment but maybe good to get it fixed for a smooth experience. I leave it to you to take a call on this. 

Otherwise, once the translation issue is fixed, everything else looks good.

donlair commented:
> If you are able to try the calypso.live link in a physical iPhone and you are able to proceed to the checkout page from the plans step, then let's move ahead. Probably what I saw was related to the iOS bug.
>
>I see a brief flicker of the user step after clicking on "Create account" that says "Your account has already been created ..."

That's an interesting issue. I see it in my local Calypso, even with the updated trunk branch, but not in production. I agree that it's probably related to the bug you surfaced.

donlair commented:
@niranjan-uma-shankar alrighty, translations look like they're available in Calypso now - did you want to take a final look at the PR?

niranjan-uma-shankar reviewed (approved):
LGTM :shipit:

niranjan-uma-shankar commented on client/signup/steps/user/index.jsx:
```diff
@@ -116,6 +120,13 @@ export class UserStep extends Component {
+			? 'registration_email_only_desktop'
+			: 'registration_email_only_mobile';
+
+		loadExperimentAssignment( experimentCheck ).then( ( experimentName ) => {
```
As per the discussion in p1637326194065900-slack-C7HH3V5AS, let's be very specific about the assignment and only assign `onboarding` flow users. Otherwise we'll add to white noise by assigning other signup flow users to the experiment.

donlair replied:
@niranjan-uma-shankar ah, we've got an `onboarding` check in the `SignupForm` but that won't prevent user assignment. I'll move that check here, good call.

donlair committed:
Limiting test assignment to onboarding flow

donlair commented:
@niranjan-uma-shankar I moved that onboarding flow check, so this is ready for another look when you get a chance.

niranjan-uma-shankar commented on client/signup/steps/user/index.jsx:
```diff
@@ -108,14 +112,24 @@ export class UserStep extends Component {
+		if ( flowName === 'onboarding' ) {
+			const experimentCheck = this.state.isDesktop
+				? 'registration_email_only_desktop'
+				: 'registration_email_only_mobile';
```
Just calling out that with this check, tablet resolutions will also get assigned to the mobile test. Maybe that's okay since tablet and mobile have similar touch screen experiences, what do you think? Or should we restrict the test to mobile only and exclude tablet?

donlair replied:
This is anecdotal, but my browsing behavior is essentially the same on tablets and mobile devices so it seems fine to group them together. I guess the exception would be something like an iPad Pro with a physical keyboard attached, but I doubt that would be a large enough group of people to skew the results.

niranjan-uma-shankar replied:
I agree, it's the same anecdotal experience for me too. I'll run a quick test of things again.

niranjan-uma-shankar commented on client/blocks/signup-form/passwordless.jsx:
```diff
@@ -0,0 +1,258 @@
+		} );
+
+		try {
+			const response = await wpcom.undocumented().usersNew(
```
Ughh, looks like https://github.com/Automattic/wp-calypso/pull/58109 removed the `usersNew()` undocumented method. So, the signup is no longer working with this method. Could you please change this to the `wpcom.req.post` method like it's used here?

https://github.com/Automattic/wp-calypso/blob/02b621f7bf67a2cdb4353bd1e05e5a9cca0a1bb9/client/lib/signup/step-actions/index.js#L660-L662

donlair committed:
Migrating to wpcom.req

donlair commented:
Good catch, I switched it over to wpcom.req and it's ready for another look @niranjan-uma-shankar

niranjan-uma-shankar reviewed (approved):
LGTM
